app/policies/article_policy.rb

```diff
@@ -0,0 +1,45 @@
+class ArticlePolicy < ApplicationPolicy
+  authorize :user, allow_nil: true
+
+  relation_scope do |relation|
+    if user.nil?
+      relation.published
+    else
+      relation.where(user_id: user.id).or(relation.published)
+    end
+  end
+
+  def index?
+    true
+  end
+
+  def create?
+    user.present?
+  end
+
+  def new?
+    create?
+  end
+
+  def show?
+    true
+  end
+
+  def destroy?
+    owner?
+  end
+
+  def edit?
+    update?
+  end
+
+  def update?
+    owner?
+  end
+
+  private
+
+  def owner?
+    user.present? && record.user == user
```

Comment:
How about?
```suggestion
    user.present? && record.user_id == user.id
```

Reply:
Yes, i wanted to do this, but in first time result was is failure, but rn this is working successfully! Nice